scripts/perftune.py

```diff
@@ -549,13 +563,27 @@ def _get_irqs(self):
         pass
 
 #### Private methods ############################
+    def __refresh_metadata_token(self):
+        self._metadata_token_time = datetime.datetime.now()
+        self._metadata_token = curl(self.EC2_METADATA_BASE_URL + "api/token", headers={"X-aws-ec2-metadata-token-ttl-seconds": self.EC2_METADATA_TOKEN_TTL}, method="PUT", timeout=0.1)
+
+    def __instance_metadata(self, path):
+        if not self._metadata_token:
+            self.__refresh_metadata_token()
+        else:
+            time_diff = datetime.datetime.now() - self._metadata_token_time
+            time_diff_sec = int(time_diff.total_seconds())
+            if time_diff_sec >= self.EC2_METADATA_TOKEN_TTL - 120:
+                self.__refresh_metadata_token()
+        return curl(self.EC2_METADATA_BASE_URL + path, headers={"X-aws-ec2-metadata-token": self._metadata_token}, timeout=0.1)
```

Comment:
Why not to hide all this "token refresh" logic (plus the memoization) inside the what is now `__refresh_metadata_token(...)`:
* Rename the method to be a property called `__metadata_token(self)`
* Put the memoization and refresh code inside the property code.

In such a case reading the metadata method is going to look at follows (I also patch it following my next comment below):

```
def __instance_metadata(self, path):
    return curl(self.EC2_METADATA_BASE_URL + "meta-data/" + path, headers={"X-aws-ec2-metadata-token": self.__metadata_token}, timeout=0.1)
```

Reply:
Fixed on the latest push.